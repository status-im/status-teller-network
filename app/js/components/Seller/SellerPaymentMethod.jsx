import React, {Component} from 'react';
import {ButtonGroup} from 'reactstrap';
import PropTypes from 'prop-types';

import CheckButton from '../ui/CheckButton';

class SellerPaymentMethod extends Component {
  togglePaymentMethod(selectedMethod) {
    this.props.togglePaymentMethod(selectedMethod);
  }

  render() {
    return (
      <React.Fragment>
        <h2>Payment methods that you want to accept</h2>

        <ButtonGroup vertical className="w-100">
          {this.props.methods.map((asset, idx) => (
            <CheckButton active={this.props.selectedMethods.indexOf(idx) > -1}
                         key={'asset-' + idx}
                         onClick={(_e) => this.togglePaymentMethod(idx)}>
              {asset}
            </CheckButton>
          ))}
        </ButtonGroup>

        {this.props.selectedMethods.length === 0 && <p className="text-info">Select one or more payment method to move to the next page</p>}
      </React.Fragment>
    );
  }
}

SellerPaymentMethod.propTypes = {
  methods: PropTypes.array,
  togglePaymentMethod: PropTypes.func,
  selectedMethods: PropTypes.array
};


export default SellerPaymentMethod;
