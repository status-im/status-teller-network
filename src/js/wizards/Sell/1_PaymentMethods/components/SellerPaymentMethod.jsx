import React, {Component} from 'react';
import {ButtonGroup} from 'reactstrap';
import PropTypes from 'prop-types';

import CheckButton from '../../../../ui/CheckButton';
import { POPULAR_PAYMENT_METHODS_INDEXES, PAYMENT_METHODS } from '../../../../features/metadata/constants';

class SellerPaymentMethod extends Component {
  togglePaymentMethod(selectedMethod) {
    this.props.togglePaymentMethod(parseInt(selectedMethod, 10));
  }

  render() {
    return (
      <React.Fragment>
        <h2 className="mb-4">Payment methods that you want to accept</h2>
        <span className="text-muted text-small">Popular</span>
        <ButtonGroup vertical className="w-100">
          {POPULAR_PAYMENT_METHODS_INDEXES.map((idx) => (
            <CheckButton active={this.props.selectedMethods.indexOf(idx) > -1}
                         key={'method-' + idx} isCheckBox
                         onClick={(_e) => this.togglePaymentMethod(idx)}>
              {PAYMENT_METHODS[idx]}
            </CheckButton>
          ))}
        </ButtonGroup>

        <span className="text-muted text-small mt-3">All payment methods (A-Z)</span>
        <ButtonGroup vertical className="w-100">
          {Object.keys(PAYMENT_METHODS).filter(x => POPULAR_PAYMENT_METHODS_INDEXES.indexOf(x) === -1).map((idx) => (
            <CheckButton active={this.props.selectedMethods.indexOf(parseInt(idx, 10)) > -1}
                         key={'method-' + idx} isCheckBox
                         onClick={(_e) => this.togglePaymentMethod(idx)}>
             {PAYMENT_METHODS[idx]}
            </CheckButton>
          ))}
        </ButtonGroup>

        {this.props.selectedMethods.length === 0 && <p className="text-muted">Select one or more payment method to move to the next page</p>}
      </React.Fragment>
    );
  }
}

SellerPaymentMethod.propTypes = {
  togglePaymentMethod: PropTypes.func,
  selectedMethods: PropTypes.array
};


export default SellerPaymentMethod;
