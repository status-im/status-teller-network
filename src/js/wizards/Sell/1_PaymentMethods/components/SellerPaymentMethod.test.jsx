import React from "react";
import { shallow } from "enzyme";

import SellerPaymentMethod from "./SellerPaymentMethod";

describe('SellerPaymentMethod', () => {
  it('should render correctly', () => {
    const component = shallow(<SellerPaymentMethod methods={[]} togglePaymentMethod={() => undefined} selectedMethods={[]}/>);
  
    expect(component).toMatchSnapshot();
  });
});
