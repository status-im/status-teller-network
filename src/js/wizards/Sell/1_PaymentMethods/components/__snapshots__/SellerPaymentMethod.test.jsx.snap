// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SellerPaymentMethod should render correctly 1`] = `
<Fragment>
  <h2
    className="mb-4"
  >
    Payment methods that you want to accept
  </h2>
  <span
    className="text-muted text-small"
  >
    Popular
  </span>
  <ButtonGroup
    className="w-100"
    role="group"
    tag="div"
    vertical={true}
  >
    <CheckButton
      active={false}
      align="right"
      isCheckBox={true}
      key="method-1"
      onClick={[Function]}
    >
      Bank transfer
    </CheckButton>
    <CheckButton
      active={false}
      align="right"
      isCheckBox={true}
      key="method-2"
      onClick={[Function]}
    >
      Cash (in person)
    </CheckButton>
    <CheckButton
      active={false}
      align="right"
      isCheckBox={true}
      key="method-3"
      onClick={[Function]}
    >
      Cash (deposit)
    </CheckButton>
  </ButtonGroup>
  <span
    className="text-muted text-small mt-3"
  >
    All payment methods (A-Z)
  </span>
  <ButtonGroup
    className="w-100"
    role="group"
    tag="div"
    vertical={true}
  >
    <CheckButton
      active={false}
      align="right"
      isCheckBox={true}
      key="method-4"
      onClick={[Function]}
    >
      AdvCash
    </CheckButton>
    <CheckButton
      active={false}
      align="right"
      isCheckBox={true}
      key="method-5"
      onClick={[Function]}
    >
      Alfa cash-in
    </CheckButton>
    <CheckButton
      active={false}
      align="right"
      isCheckBox={true}
      key="method-6"
      onClick={[Function]}
    >
      Alipay
    </CheckButton>
    <CheckButton
      active={false}
      align="right"
      isCheckBox={true}
      key="method-7"
      onClick={[Function]}
    >
      Cardless cash
    </CheckButton>
    <CheckButton
      active={false}
      align="right"
      isCheckBox={true}
      key="method-8"
      onClick={[Function]}
    >
      Efecty
    </CheckButton>
    <CheckButton
      active={false}
      align="right"
      isCheckBox={true}
      key="method-9"
      onClick={[Function]}
    >
      Gift card
    </CheckButton>
    <CheckButton
      active={false}
      align="right"
      isCheckBox={true}
      key="method-10"
      onClick={[Function]}
    >
      IMPS
    </CheckButton>
    <CheckButton
      active={false}
      align="right"
      isCheckBox={true}
      key="method-11"
      onClick={[Function]}
    >
      Interac e-Transfer
    </CheckButton>
    <CheckButton
      active={false}
      align="right"
      isCheckBox={true}
      key="method-12"
      onClick={[Function]}
    >
      International wire
    </CheckButton>
    <CheckButton
      active={false}
      align="right"
      isCheckBox={true}
      key="method-13"
      onClick={[Function]}
    >
      M-PESA
    </CheckButton>
    <CheckButton
      active={false}
      align="right"
      isCheckBox={true}
      key="method-14"
      onClick={[Function]}
    >
      Mercado Pago
    </CheckButton>
    <CheckButton
      active={false}
      align="right"
      isCheckBox={true}
      key="method-15"
      onClick={[Function]}
    >
      MoneyGram
    </CheckButton>
    <CheckButton
      active={false}
      align="right"
      isCheckBox={true}
      key="method-16"
      onClick={[Function]}
    >
      PAYEER
    </CheckButton>
    <CheckButton
      active={false}
      align="right"
      isCheckBox={true}
      key="method-17"
      onClick={[Function]}
    >
      PayNow
    </CheckButton>
    <CheckButton
      active={false}
      align="right"
      isCheckBox={true}
      key="method-18"
      onClick={[Function]}
    >
      PayPal
    </CheckButton>
    <CheckButton
      active={false}
      align="right"
      isCheckBox={true}
      key="method-19"
      onClick={[Function]}
    >
      PayTM
    </CheckButton>
    <CheckButton
      active={false}
      align="right"
      isCheckBox={true}
      key="method-20"
      onClick={[Function]}
    >
      Pingit
    </CheckButton>
    <CheckButton
      active={false}
      align="right"
      isCheckBox={true}
      key="method-21"
      onClick={[Function]}
    >
      QIWI
    </CheckButton>
    <CheckButton
      active={false}
      align="right"
      isCheckBox={true}
      key="method-22"
      onClick={[Function]}
    >
      SEPA transfer
    </CheckButton>
    <CheckButton
      active={false}
      align="right"
      isCheckBox={true}
      key="method-23"
      onClick={[Function]}
    >
      Skrill
    </CheckButton>
    <CheckButton
      active={false}
      align="right"
      isCheckBox={true}
      key="method-24"
      onClick={[Function]}
    >
      Swish
    </CheckButton>
    <CheckButton
      active={false}
      align="right"
      isCheckBox={true}
      key="method-25"
      onClick={[Function]}
    >
      TransferWise
    </CheckButton>
    <CheckButton
      active={false}
      align="right"
      isCheckBox={true}
      key="method-26"
      onClick={[Function]}
    >
      UPI
    </CheckButton>
    <CheckButton
      active={false}
      align="right"
      isCheckBox={true}
      key="method-27"
      onClick={[Function]}
    >
      Venmo
    </CheckButton>
    <CheckButton
      active={false}
      align="right"
      isCheckBox={true}
      key="method-28"
      onClick={[Function]}
    >
      WebMoney
    </CheckButton>
    <CheckButton
      active={false}
      align="right"
      isCheckBox={true}
      key="method-29"
      onClick={[Function]}
    >
      WeChat Pay
    </CheckButton>
    <CheckButton
      active={false}
      align="right"
      isCheckBox={true}
      key="method-30"
      onClick={[Function]}
    >
      Western Union
    </CheckButton>
    <CheckButton
      active={false}
      align="right"
      isCheckBox={true}
      key="method-31"
      onClick={[Function]}
    >
      Yandex.Money
    </CheckButton>
    <CheckButton
      active={false}
      align="right"
      isCheckBox={true}
      key="method-99999"
      onClick={[Function]}
    >
      Other
    </CheckButton>
  </ButtonGroup>
  <p
    className="text-muted"
  >
    Select one or more payment method to move to the next page
  </p>
</Fragment>
`;
